sdk/database/dbplugin/v5/plugin_client.go

```diff
@@ -3,59 +3,46 @@ package dbplugin
 import (
 	"context"
 	"errors"
-	"sync"
 
 	log "github.com/hashicorp/go-hclog"
 	plugin "github.com/hashicorp/go-plugin"
+	"github.com/hashicorp/vault/sdk/database/dbplugin/v5/proto"
 	"github.com/hashicorp/vault/sdk/helper/pluginutil"
 )
 
-// DatabasePluginClient embeds a databasePluginRPCClient and wraps it's Close
-// method to also call Kill() on the plugin.Client.
 type DatabasePluginClient struct {
-	client *plugin.Client
-	sync.Mutex
-
+	client pluginutil.Multiplexer
 	Database
 }
 
 // This wraps the Close call and ensures we both close the database connection
 // and kill the plugin.
 func (dc *DatabasePluginClient) Close() error {
 	err := dc.Database.Close()
-	dc.client.Kill()
+	dc.client.Close()
 
 	return err
 }
 
+// pluginSets is the map of plugins we can dispense.
+var PluginSets = map[int]plugin.PluginSet{
+	5: {
+		"database": &GRPCDatabasePlugin{multiplexingSupport: false},
+	},
+	6: {
+		"database": &GRPCDatabasePlugin{multiplexingSupport: true},
+	},
+}
+
 // NewPluginClient returns a databaseRPCClient with a connection to a running
-// plugin. The client is wrapped in a DatabasePluginClient object to ensure the
-// plugin is killed on call of Close().
+// plugin.
 func NewPluginClient(ctx context.Context, sys pluginutil.RunnerUtil, pluginRunner *pluginutil.PluginRunner, logger log.Logger, isMetadataMode bool) (Database, error) {
-	// pluginSets is the map of plugins we can dispense.
-	pluginSets := map[int]plugin.PluginSet{
-		5: {
-			"database": new(GRPCDatabasePlugin),
-		},
-	}
-
-	client, err := pluginRunner.RunConfig(ctx,
-		pluginutil.Runner(sys),
-		pluginutil.PluginSets(pluginSets),
-		pluginutil.HandshakeConfig(handshakeConfig),
-		pluginutil.Logger(logger),
-		pluginutil.MetadataMode(isMetadataMode),
-		pluginutil.AutoMTLS(true),
-	)
+	pluginClient, err := sys.NewPluginClient(ctx, pluginRunner, logger, isMetadataMode)
```

Comment:
How does `NewPluginClient` handle non-multiplexed client? `sys.NewPluginClient` seems to only return clients that are multiplexed.